### PR TITLE
[Auto3DSeg] Add mlflow support in autorunner.

### PR DESCRIPTION
**Pull request by dongyang0122**
Add MLflow support in AutoRunner Class.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/auto3dseg/auto_runner.py [hunk 1]
```diff
@@ -209,6 +211,7 @@ def __init__(
         not_use_cache: bool = False,
         templates_path_or_url: str | None = None,
         allow_skip: bool = True,
+        mlflow_tracking_uri: str | None = None,
```
Comment (drbeh) [inline monai/apps/auto3dseg/auto_runner.py:214]: I think having a method `set_mlflow_params(mlflow_tracking_uri=None, **kwargs)` might be better as we may need to modify more parameters of mlflow?

File: monai/apps/auto3dseg/bundle_gen.py [hunk 2]
```diff
@@ -129,6 +130,17 @@ def set_data_source(self, data_src_cfg: str) -> None:
         """
         self.data_list_file = data_src_cfg
 
+    def set_mlflow_tracking_uri(self, mlflow_tracking_uri: str | None = None) -> None:
```
Comment (drbeh) [inline monai/apps/auto3dseg/bundle_gen.py:133]: The `__init__` is already accepting `mlflow_tracking=_uri`, do we really need this method here? I believe we need either this or that (the same for AutoRunner), what do you think?

**Comment by myron** [comment]
I think this should be a part of kwargs of Autorunner, instead of adding more names parameters for very specific use case. 

Then you check for it as 
If kwargs.get('mlflow_url', None) is not None:

**Comment by drbeh** [comment]
> I think this should be a part of kwargs of Autorunner, instead of adding more names parameters for very specific use case.
> 
> Then you check for it as If kwargs.get('mlflow_url', None) is not None:

@myron @dongyang0122 As the zen of python says "explicit is better than implicit". In my opinion, having more names means better documentation. Deferring to kwargs usually makes sense when dealing with a third party parameters that might change over time without out notice, but mlflow_tracking_uri is our parameter and we decide how to use it (although we are passing it to mlflow).

File: monai/apps/auto3dseg/bundle_gen.py (hunk 2 above)
Comment (dongyang0122) [reply monai/apps/auto3dseg/bundle_gen.py:133]: I think that the modification of more parameters of mlflow would be better.

File: monai/apps/auto3dseg/auto_runner.py (hunk 1 above)
Comment (dongyang0122) [reply monai/apps/auto3dseg/auto_runner.py:214]: Can you provide a few examples about which params can be modified?

File: monai/apps/auto3dseg/bundle_gen.py (hunk 2 above)
Comment (dongyang0122) [reply monai/apps/auto3dseg/bundle_gen.py]: But we need to define the scope of the parameters to cover in the first place.

Comment (dongyang0122) [reply monai/apps/auto3dseg/bundle_gen.py]: Having this function follows conventions as other similar functions in `BundleAlgo` and `BundleGen`, for example, `set_data_stats`.

Comment (drbeh) [reply monai/apps/auto3dseg/bundle_gen.py]: I agree with you and I like this method but I believe we don't need the redundancy of having both this method and a parameter in the init method. If we keep this, we may need to remove `mlflow_tracking_uri` from __init__. What do you think?

File: monai/apps/auto3dseg/auto_runner.py (hunk 1 above)
Comment (dongyang0122) [reply monai/apps/auto3dseg/auto_runner.py:214]: Okay.

File: monai/apps/auto3dseg/bundle_gen.py (hunk 2 above)
Comment (dongyang0122) [reply monai/apps/auto3dseg/bundle_gen.py]: I follow the design of `data_stats_filename` and `data_src_cfg_name` in the same class. @mingxin-zheng could you please share some thoughts here? Thanks!

**Comment by dongyang0122** [comment]
> > I think this should be a part of kwargs of Autorunner, instead of adding more names parameters for very specific use case.
> > Then you check for it as If kwargs.get('mlflow_url', None) is not None:
> 
> @myron @dongyang0122 As the zen of python says "explicit is better than implicit". In my opinion, having more names means better documentation. Deferring to kwargs usually makes sense when dealing with a third party parameters that might change over time without our notice, but mlflow_tracking_uri is our parameter and we decide how to use it (although we are passing it to mlflow).

I acknowledge the perspectives presented from both sides. Incorporating an input variable `mlflow_tracking_uri` would enhance parameter visibility and documentation, whereas including it under ‘kwargs’ could conserve script space. Personally, I favor the first approach, though both methods are viable. Input from individuals @Nic-Ma and @daguangxu on this matter would be beneficial to finalize the decision.

**Comment by Nic-Ma** [comment]
Hi @drbeh , @dongyang0122 ,

Do we have any usecase to allow users to set `experiment_name`? If no, I would vote for current implementation with `mlflow_tracking_uri` arg directly.

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**